### ISSUE TITLE
Cannot set property request of #<Socket> which has only a getter

### ISSUE THREAD

**Issue by IceDragon200** [socketio/socket.io#2155]
tl;dr I needed ES6 for a project, some rather arcane code in Socket is causing me unrest.

Running iojs v2.3.0 with --strict_mode, spat this error out:
```
<REDACTED>/chat-server/node_modules/socket.io/lib/socket.js:63
  this.request = client.request;
TypeError: Cannot set property request of #<Socket> which has only a getter
```

From what I've read in the source, Socket uses `__defineGetter__` to define the request attribute reader, which just returns `this.conn.request`, `this.conn` was set from `client.conn`, and `client.request` is just a shortcut for `client.conn.request`, so why for the love of code is `request` being set, when its reader is being delegated to `conn.request` via `__defineGetter__` anyway!

This is what is happening: (I'm using get/set here for illustration)
```javascript
setRequest(request) {
  this.request = request;
}

getRequest() {
  // So, whatever happened to `this.request`
  return this.conn.request;
}

setRequest(myRequest); 
getRequest() == myRequest; // false
```

**Comment by nkzawa** [comment]
related: https://github.com/Automattic/socket.io/pull/2136

**Comment by benjaminbalazs** [comment]
No matter what I do, I can't get rid of this issue. Latest node, latest socket.io harmony and use-strict on. Any idea?

**Comment by mastermind202** [comment]
@benjaminbalazs: I'm running into the same issue running node v4.1.1 and socket.io 1.3.7.
Did you ever find an elegant solution to this problem?

**Comment by kevinhikaruevans** [comment]
:+1:

Happening to me as well. Was passing in `--use-strict --harmony --harmony_scoping`

**Comment by mastermind202** [comment]
Like @nkzawa pointed out, commenting out the offending line (63) in `node_modules/socket.io/lib/socket.js` fixes the issue without any ill effects that I've noticed.

It would be great if it was fixed in the actual socket.io package.

**Comment by benjaminbalazs** [comment]
@mastermind202 I think I had to turn off all those mentioned flags, but to be honest not sure. At that time I left it not working and just recently I installed socket.io and it worked just fine.

One more note though, make sure you connect or test the connection with socket.io's client, otherwise it won't work properly.

Commenting out the socket.js was not an option for me as I do npm install on deploy.

**Comment by ajcrites** [comment]
I'm also experiencing this issue. A built-in resolution would be great.

**Comment by alarner** [comment]
:+1: happening for me as well with --use-strict

**Comment by nkzawa** [comment]
Fixed in 1.4
